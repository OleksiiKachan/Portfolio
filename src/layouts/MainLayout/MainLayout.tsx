import React, { Component, ReactNode } from 'react';
import classNames from 'classnames';
import HeaderContainer from '../../containers/HeaderContainer';
import './MainLayout.scss';

/*
  <MainLayout
    className=''
  >
    {children}
  </MainLayout>
*/

type PropsType = {
  children: ReactNode;
  className?: string;
};

class MainLayout extends Component<PropsType> {
  render() {
    return (
      <div className={classNames('app')}>
        <HeaderContainer />
        <main className={classNames('app__mainContent', this.props.className)}>
          {this.props.children}
        </main>
      </div>
    );
  }
}

export default MainLayout;
