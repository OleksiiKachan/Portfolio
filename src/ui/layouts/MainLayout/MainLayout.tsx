import React, { Component, ReactNode } from 'react';
import classNames from 'classnames';

import { HeaderContainer, FooterContainer } from '../../pages';
import { GlobalStyle } from '../../assets/style';

/*
  <MainLayout
    className=''
  >
    {children}
  </MainLayout>
*/

type PropsType = {
  children: ReactNode;
  className?: string;
};

class MainLayout extends Component<PropsType> {
  render() {
    return (
      <div className={classNames('app')}>
        <GlobalStyle />
        <HeaderContainer />
        <main className={classNames('app__mainContent', this.props.className)}>
          {this.props.children}
        </main>
        <FooterContainer />
      </div>
    );
  }
}

export default MainLayout;
