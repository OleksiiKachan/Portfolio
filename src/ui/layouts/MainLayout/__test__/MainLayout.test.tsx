import React from 'react';
import { shallow } from 'enzyme';
import MainLayout from '../MainLayout';

describe('MainLayout', () => {
  it('Snapshot', () => {
    const component = shallow(<MainLayout>content</MainLayout>);
    expect(component).toMatchSnapshot();
  });
});
