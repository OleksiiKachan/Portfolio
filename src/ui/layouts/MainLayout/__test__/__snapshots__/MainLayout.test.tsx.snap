// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout Snapshot 1`] = `
<div
  className="app"
>
  <GlobalStyleComponent />
  <withRouter(HeaderContainer) />
  <main
    className="app__mainContent"
  >
    content
  </main>
  <FooterContainer />
</div>
`;
